### DISCUSSION TITLE
What's the difference between turbo prune web & turbo build and turbo build --filter=web ?

### DISCUSSION THREAD

**Discussion by MattinaYang** [vercel/turborepo#9496] [Help] [answered]
### Summary

There're two ways to build project under apps folder, for example, the project's name is web:
1. turbo build --filter=web
2. turbo prune web && cd web && turbo build
I'm wondering to build web in my docker, after reading the doc, I found `turbo prune web --docker && cd web && turbo build` is the best way. But I wanna know the difference between the two ways. Anyone help?

**Comment by anthonyshew** [comment, marked as answer]
`turbo prune` helps to creates a smaller subset of the Workspace to work from for your Docker container while `--filter` only executes a subset of tasks. `turbo prune` is a smaller starting point for the tasks that you're going to eventually run.

 A larger description of the process can be found here: https://turbo.build/repo/docs/guides/tools/docker

**Reply by MattinaYang** [reply]
Thanks for your reply :)  But according this Dockerfile in examples (https://github.com/vercel/turborepo/blob/main/examples/with-docker/apps/web/Dockerfile), I'm confused why in `yarn turbo build --filter=web `, the `--filter=web` is needed while `turbo prune web --docker` has ran before it.  I mean, after running `prune`, the folder should has no other things unrelated with web? So is there no need to add `--filter=web`?

**Reply by anthonyshew** [reply]
Correct, it isn't needed. https://github.com/vercel/turborepo/pull/9513

Sorry for the confusion.
